src/java/org/apache/cassandra/config/InheritingClass.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.config;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.cassandra.exceptions.ConfigurationException;
+
+public class InheritingClass extends ParameterizedClass
+{
+    public String inherits = null;
+
+    public InheritingClass()
+    {
+    }
+
+    public InheritingClass(String inherits, String class_name, Map<String, String> parameters)
+    {
+        super(class_name, parameters);
+        this.inherits = inherits;
+    }
+
+    public InheritingClass(Map<String, ?> p)
```

Comment:
nit: either remove or throw `@SuppressWarnings("unused")` on this one and the no-arg actor (which I'm guessing is there for framework reasons)